src/kernels/variables/types.ts

```diff
@@ -56,6 +57,7 @@ export interface IJupyterVariables {
     ): Promise<IJupyterVariable | undefined>;
     // This is currently only defined in kernelVariables.ts
     getVariableProperties?(name: string, kernel?: IKernel, cancelToken?: CancellationToken): Promise<JSONObject>;
+    evaluate(code: string, frameId?: number): Promise<DebugProtocol.EvaluateResponse['body']>;
```

Comment:
I don't think `IJuptyerVariables1` needs such a generic method.
Basically we're exposing the ability to execute random code in an interface used specifically to get variables.
Also `frameId` is specific to debugger, hence this doesn't belong here.

Reply:
Maybe we could make a more generic service that you can execute a piece of code on. The 'executeSilently' method we use elsewhere seems like it would be a good candidate. It could keep track of the topmost frame id and use the debugger if the kernel was 'frozen' at a breakpoint.